### PR TITLE
[XLA:CPU][oneDNN] Fix oneDNN matmul test timeout

### PR DESCRIPTION
**Pull request by akhilgoe**
This PR addresses the test timeout observed in oneDNN matmul test file.

In particular, this PR:

1. Replaces the test titled ConsecutiveBinaryAdd with a smaller test such that the test still hits the targeted failure case.
2. Shards the test file.
3. In addition, this PR also replaces all instances of the old proto definitions with the new ones.

### CODE REVIEW

**Comment by thomasjoerg** [comment]
I'm not the best person to review this change. @ezhulenev is dispatching PRs to reviewers this week.

File: xla/service/cpu/tests/onednn_matmul_test.cc [hunk 1]
```diff
@@ -1496,44 +1498,18 @@ TEST_F(MatmulTest, WeightsPrepackAndScratch) {
...
-    dot.20 = f32[1000000,3] dot(broadcast.3, constant.9), lhs_contracting_dims={}, rhs_contracting_dims={}
-    add.21 = f32[1000000,3] add(add.19, dot.20)
-    constant.6 = f32[] constant(0)
-    reduce.26 = f32[3] reduce(add.21, constant.6), dimensions={0}, to_apply=region_0.22
-    reshape.27 = f32[1,3] reshape(reduce.26)
-    negate.28 = f32[1,3] negate(reshape.27)
-    ROOT reduce.33 = f32[3] reduce(negate.28, constant.6), dimensions={0}, to_apply=region_1.29
+  ENTRY matmul.test.f32 {
+    arg0.1 = f32[128,32,4,4] parameter(0), parameter_replication={false}
+    arg0.2 = f32[128,32,4,4] parameter(1), parameter_replication={false}
```
Comment (penpornk) [inline xla/service/cpu/tests/onednn_matmul_test.cc:1503]: Is `parameter_replication` needed? If not, let's remove it?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the fixes!

File: xla/service/cpu/tests/onednn_matmul_test.cc (hunk 1 above)
Comment (akhilgoe) [reply xla/service/cpu/tests/onednn_matmul_test.cc:1503]: Thanks for the review! Yes, the test does not require that field. Removed it in the latest commit.

**Review by penpornk — APPROVED** [review]
Thank you!

**Pull request closed without merging**